### PR TITLE
PLT-558 Right after Vasil HF on Cardano mainnet, merge next-node into main

### PR DESCRIPTION
**Pull request by berewt**
It's technically a merge of the most recent version of main into next-node.

Notable impact:
- A few errors have changed, as we have enabled the `cardano-ledger` validation in next-node
- Marconi code should no handle babbage era Tx, and some copy/paste needed because of the old version of `cardano-node` used in `main` has been removed
- Some tests that were enabled as they can now run correctly thanks to `cardano-ledger` validation.
- `mkTxInfoVx` functions and the generators that relied on them were removed, as they used our custom validation rules.
- `MustProduceAtLeast` tests are disabled as they don't run on Babbage era atm.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

**Review by sjoerdvisscher — APPROVED** [review]
💯

File: plutus-ledger/src/Ledger/Generators.hs [hunk 1]
```diff
@@ -135,7 +136,7 @@ generatorModel =
 --   plutus-ledger (in particular, 'Ledger.Tx.unspentOutputs') we note the
 --   unspent outputs of the chain when it is first created.
 data Mockchain = Mockchain {
-    mockchainInitialTxPool :: [CardanoTx],
+    mockchainInitialTxPool :: [Tx],
```
[the remark this answers is not in the document]
Comment (berewt) [reply plutus-ledger/src/Ledger/Generators.hs:139]: Oh you must be right. Oh did I miss that?!

[the remark this answers is not in the document]
Comment (berewt) [reply plutus-ledger/src/Ledger/Generators.hs:139]: Although, we were generating `EmulatorTx`. Anyway, I need to revert the signature and adapt the code accordingly.

[the remark this answers is not in the document]
Comment (sjoerdvisscher) [reply plutus-ledger/src/Ledger/Generators.hs]: Yes, it should be `CardanoTx`, although it's not very important.

[the remark this answers is not in the document]
Comment (berewt) [reply plutus-ledger/src/Ledger/Generators.hs]: It should be fixed, Evgenii is right, it may help when/if we'll want to remove `Tx`.

**Comment by eyeinsky** [comment]
Removed some more copy-pasted code from marconi/test/Spec.hs and made `extra-source-files` a package level field (it gave an unrecognized field warning otherwise) in the last two commits of this branch: https://github.com/input-output-hk/plutus-apps/commits/ml/merge-main

**Comment by eyeinsky** [comment]
The main branch has also progressed a bit, especially the plutus-streaming => cardano-streaming rename will create some conflict.

**Comment by berewt** [comment]
@eyeinsky Did you check if the tests passed after you remove the duplication of `genProtocolParameters`? When I tried to rewove it, I had an issue in the roundtrip test when `protocolParamDecentralization` was set to `Nothing` by the generator.

**Comment by eyeinsky** [comment]
It's funny that it does pass for me, but it shouldn't (?)

**Comment by berewt** [comment]
Well. I've rebase my branch with your changes and it passes, I guess I did something wrong elsewhere first, corrected it and that the generator issue was just a workaround that is no longer necessary. Thanks for pointing it out!

**Review by eyeinsky — APPROVED** [review]

**Pull request merged**